modules/microvm/common/storagevm.nix

```diff
@@ -111,29 +125,64 @@ in
         }
       ];
 
-      environment.persistence.${cfg.mountPath} = mkMerge [
-        {
-          hideMounts = true;
-          directories = [
-            "/var/lib/nixos"
-          ];
-          files = [
-            "/etc/ssh/ssh_host_ed25519_key.pub"
-            "/etc/ssh/ssh_host_ed25519_key"
-          ];
-        }
-        { inherit (cfg) directories users files; }
-        (mkIf config.ghaf.users.loginUser.enable {
-          directories = [
-            "/var/lib/systemd/home"
-          ];
-        })
-      ];
+      preservation = {
+        enable = true;
+        preserveAt.${cfg.mountPath} = mkMerge [
+
+          # Standard directories and files
+          {
+            directories = [
+              "/var/lib/nixos"
+            ];
+            files = [
+              {
+                file = "/etc/machine-id";
+                inInitrd = true;
+              }
+            ];
+          }
+
+          # User-specific directories and files
+          { inherit (cfg) directories users files; }
+
+          # Optional log preservation
+          (mkIf cfg.preserveLogs {
+            directories =
+              [
+                "/var/log/journal"
+              ]
+              ++ optionals config.security.auditd.enable [
+                "/var/log/audit"
+              ];
+          })
 
-      # Workaround, fixes homed machine-id dependency
-      environment.etc = lib.optionalAttrs config.ghaf.users.loginUser.enable {
-        machine-id.text = "d8dee68f8d334c79ac8f8229921e0b25";
+          # Optional files for ssh
+          (mkIf (config.services.sshd.enable && config.ghaf.profiles.debug.enable) {
```

Comment:
sshd.enable is probably enough. I am not sure we will be able to remove all ssh for release. and then the ssh can be gated to be on or off through debug if needed